plugins/Overlay/Overlay.php

```diff
@@ -37,4 +41,39 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = 'General_OverlayRowActionTooltipTitle';
         $translationKeys[] = 'General_OverlayRowActionTooltip';
     }
+
+    /**
+     * Returns if a request belongs to the Overlay page
+     *
+     * Whenever we change the Overlay, or any feature that is available on that page, this list needs to be adjusted
+     * Otherwise it can happen, that the session cookie is sent with samesite=lax, which might break the session in Overlay
+     * See https://github.com/matomo-org/matomo/pull/18648
+     */
+    public static function isOverlayRequest()
+    {
+        $module = Piwik::getModule();
+        $action = Piwik::getAction();
+        $method = Common::getRequestVar('method', '', 'string');
+
+        $isOverlay = $module == 'Overlay';
+        $referrerUrlQuery = parse_url(Url::getReferrer() ?? '', PHP_URL_QUERY);
```

Comment:
just to be safe we could additionally get the host of the referrer, and validate it's a valid host

Reply:
Added an additional check for valid hosts

Reply:
👍